### PR TITLE
start springboot api

### LINKED ISSUE #1: 1 - Start a spring boot api
Create a fake crud for users
- get
- get-by-id
- post
- put
- delete will return "The id: $usersId was deleted"

Obs: Do not need to configure database in this step, else you need to avoid the configuration, for more information read the link below.
[DISABLE SPRING DATA AUTO CONFIG](https://www.baeldung.com/spring-data-disable-auto-config)

### PR DESCRIPTION
**Pull request by soaresluana**

### CODE REVIEW

File: HELP.md
Comment (camilomelges) [inline HELP.md:1]: Here you need to provide a full documentation about how to run the application and use it. Please follow these examples:
- https://github.com/camilomelges/softplan-desafio-fullstack/blob/master/README.MD
- https://github.com/camilomelges/softplan-test/blob/master/disknfiles/README.md
- https://github.com/camilomelges/softplan-test/blob/master/starwarscharacters/README.md

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		<version>3.0.6</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.camilomelges</groupId>
+	<artifactId>usersapi</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>usersapi</name>
+	<description>An API to manage users</description>
+	<properties>
+		<java.version>17</java.version>
```
Comment (camilomelges) [inline pom.xml]: change it to java version 11

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+@RestController(value = "users")
+@RequestMapping("users")
+public class UserController {
+
+  @GetMapping
+  public List<String> getAll() {
+      return List.of("1","2","3","4");
+  }
+
+  @RequestMapping(value = "{id}", method = RequestMethod.GET)
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Change it to a new implementation of Request mapping for get methods, @GetMapping

[READ](https://www.baeldung.com/spring-new-requestmapping-shortcuts)

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public String getById(@PathVariable("id") final String id) {
+        return id;
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Change it to a new implementation of Request mapping for get methods, @PostMapping

[READ](https://www.baeldung.com/spring-new-requestmapping-shortcuts)

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public Map<String, String> post(@RequestBody final Map<String, String> user) {
+        user.put("id", RandomStringUtils.randomAlphabetic(1));
+        return user;
+    }
+
+    @RequestMapping(method = RequestMethod.PUT)
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Change it to a new implementation of Request mapping for get methods, @PutMapping

[READ](https://www.baeldung.com/spring-new-requestmapping-shortcuts)

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public Map<String, String> put(@RequestBody final Map<String, String> user) {
+        return user;
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.DELETE)
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Change it to a new implementation of Request mapping for get methods, @DeleteMapping

[READ](https://www.baeldung.com/spring-new-requestmapping-shortcuts)

**Review by camilomelges — CHANGES_REQUESTED** [review]
Take a look on the comments please.

File: HELP.md
Comment (soaresluana) [reply HELP.md:1]: done

File: pom.xml (hunk 1 above)
Comment (soaresluana) [reply pom.xml]: :)

File: src/main/java/com/camilomelges/usersapi/UserController.java (hunk 2 above)
Comment (soaresluana) [reply src/main/java/com/camilomelges/usersapi/UserController.java]: done

File: src/main/java/com/camilomelges/usersapi/UserController.java (hunk 3 above)
Comment (soaresluana) [reply src/main/java/com/camilomelges/usersapi/UserController.java]: done

File: src/main/java/com/camilomelges/usersapi/UserController.java (hunk 4 above)
Comment (soaresluana) [reply src/main/java/com/camilomelges/usersapi/UserController.java]: done

File: src/main/java/com/camilomelges/usersapi/UserController.java (hunk 5 above)
Comment (soaresluana) [reply src/main/java/com/camilomelges/usersapi/UserController.java]: done

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        return id;
+    }
+
+    @PostMapping("post")
+    public Map<String, String> post(@RequestBody final Map<String, String> user) {
+        user.put("id", RandomStringUtils.randomAlphabetic(1));
+        return user;
+    }
+
+    @PutMapping("put")
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Doesn't need the "put" here, the @PutMethod already implies this.

File: src/main/java/com/camilomelges/usersapi/UserController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    public List<String> getAll() {
+        return List.of("1", "2", "3", "4");
+    }
+
+    @GetMapping("{id}")
+    public String getById(@PathVariable("id") final String id) {
+        return id;
+    }
+
+    @PostMapping("post")
```
Comment (camilomelges) [inline src/main/java/com/camilomelges/usersapi/UserController.java]: Doesn't need the "post" here, the @PostMethod already implies this.

**Review by camilomelges — APPROVED** [review]

**Pull request merged**